### PR TITLE
[PLGN-180] Cisco Umbrella Reporting - Updated plugin to use API v2

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Updated plugin to use API v2

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by igorski-r7** [comment]
> @igorski-r7 , as discussed, do you mind adding in the pagination for the records? Thanks :)

Yes, I will add it 😄

File: plugins/cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/constats.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+LIMIT_DEFAULT_VALUE = 0
```
Comment (dlaverty-r7) [inline plugins/cisco_umbrella_reporting/icon_cisco_umbrella_reporting/util/constats.py:1]: Typo in filename -> constants.py?

**Review by dlaverty-r7 — COMMENTED** [review]
Generally LGTM :) . Just a general comment - is it worth adding something in the Troubleshooting section or elsewhere to state where and what type of API key is needed? Also worth stating that legacy keys can no longer used - there is a reference to legacy keys in the reference section. Ensure the links in the references section are for the correct API etc.

File: plugins/cisco_umbrella_reporting/help.md [hunk 2]
```diff
@@ -9,8 +9,11 @@ This plugin utilizes Cisco Umbrella to get the most complete view of the relatio
 
 # Requirements
 
-* Requires an API Key and Secret from Cisco Umbrella (Legacy API Key) - In the links section it's described how to create/refresh/delete it)
-* Requires Cisco Umbrella organization ID
+* Requires an API Key and Secret from Cisco Umbrella (API Key) - In the links section it's described how to create/refresh/delete it)
```
Comment (dlaverty-r7) [inline plugins/cisco_umbrella_reporting/help.md:12]: Suggested reword- 'Refer to the links section for information on how to create/refresh/delete the API key.

File: plugins/cisco_umbrella_reporting/help.md [hunk 3]
```diff
@@ -98,27 +122,47 @@ _This plugin does not contain any triggers._
...
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|Label|string|False|The threat label|
+|Type|string|False|Type of threat|
+
 
 ## Troubleshooting
 
-_This plugin does not contain any troubleshooting information._
+* The Cisco Umbrella Legacy API Keys are no longer used. Please use the Cisco Umbrella API Keys. In the links section it's described how to create/refresh/delete it.
```
Comment (dlaverty-r7) [inline plugins/cisco_umbrella_reporting/help.md:161]: Suggested reword -> 'Refer to the links section for information on how to create/refresh/delete the API key.'

File: plugins/cisco_umbrella_reporting/help.md [hunk 4]
```diff
@@ -98,27 +122,47 @@ _This plugin does not contain any triggers._
...
 
 # Version History
 
+* 2.0.0 - Updated plugin to use API v2
```
Comment (dlaverty-r7) [inline plugins/cisco_umbrella_reporting/help.md:165]: Suggestion - Add an additional note to say that OAuth authentication is now used and organization id is no longer required to setup the connection.

**Review by dlaverty-r7 — COMMENTED** [review]
Lgtm - just a couple of small updates if you don't mind making them before merging please?

**Review by llaszuk-r7 — APPROVED** [review]

**Review by dlaverty-r7 — APPROVED** [review]
Looks good, thanks for making the suggested updates!

**Pull request merged**